To: vim_dev@googlegroups.com
Subject: Patch 7.3.176
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.176
Problem:    Ruby linking doesn't work properly on Mac OS X.
Solution:   Fix the configure check for Ruby. (Bjorn Winckler)
Files:	    src/configure.in, src/auto/configure


*** ../vim-7.3.175/src/configure.in	2011-05-05 17:23:58.000000000 +0200
--- src/configure.in	2011-05-05 18:03:38.000000000 +0200
***************
*** 1387,1396 ****
  AC_MSG_RESULT($enable_rubyinterp)
  if test "$enable_rubyinterp" = "yes" -o "$enable_rubyinterp" = "dynamic"; then
    AC_MSG_CHECKING(--with-ruby-command argument)
    AC_ARG_WITH(ruby-command, [  --with-ruby-command=RUBY  name of the Ruby command (default: ruby)],
! 	RUBY_CMD="$withval"; AC_MSG_RESULT($RUBY_CMD),
  	RUBY_CMD="ruby"; AC_MSG_RESULT(defaulting to $RUBY_CMD))
-   AC_SUBST(vi_cv_path_ruby)
    AC_PATH_PROG(vi_cv_path_ruby, $RUBY_CMD)
    if test "X$vi_cv_path_ruby" != "X"; then
      AC_MSG_CHECKING(Ruby version)
--- 1387,1396 ----
  AC_MSG_RESULT($enable_rubyinterp)
  if test "$enable_rubyinterp" = "yes" -o "$enable_rubyinterp" = "dynamic"; then
    AC_MSG_CHECKING(--with-ruby-command argument)
+   AC_SUBST(vi_cv_path_ruby)
    AC_ARG_WITH(ruby-command, [  --with-ruby-command=RUBY  name of the Ruby command (default: ruby)],
! 	RUBY_CMD="$withval"; vi_cv_path_ruby="$withval"; AC_MSG_RESULT($RUBY_CMD),
  	RUBY_CMD="ruby"; AC_MSG_RESULT(defaulting to $RUBY_CMD))
    AC_PATH_PROG(vi_cv_path_ruby, $RUBY_CMD)
    if test "X$vi_cv_path_ruby" != "X"; then
      AC_MSG_CHECKING(Ruby version)
***************
*** 1412,1429 ****
  	  RUBY_LIBS="$rubylibs"
  	fi
  	librubyarg=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG[["LIBRUBYARG"]])'`
! 	if test -f "$rubyhdrdir/$librubyarg"; then
! 	  librubyarg="$rubyhdrdir/$librubyarg"
! 	else
! 	  rubylibdir=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG[["libdir"]])'`
! 	  if test -f "$rubylibdir/$librubyarg"; then
! 	    librubyarg="$rubylibdir/$librubyarg"
! 	  elif test "$librubyarg" = "libruby.a"; then
! 	    dnl required on Mac OS 10.3 where libruby.a doesn't exist
! 	    librubyarg="-lruby"
! 	  else
! 	    librubyarg=`$vi_cv_path_ruby -r rbconfig -e "print '$librubyarg'.gsub(/-L\./, %'-L#{Config.expand(Config::CONFIG[\"libdir\"])}')"`
! 	  fi
  	fi
  
  	if test "X$librubyarg" != "X"; then
--- 1412,1426 ----
  	  RUBY_LIBS="$rubylibs"
  	fi
  	librubyarg=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG[["LIBRUBYARG"]])'`
! 	librubya=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG[["LIBRUBY_A"]])'`
!   rubylibdir=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG[["libdir"]])'`
! 	if test -f "$rubylibdir/$librubya"; then
! 	  librubyarg="$librubyarg"
!     RUBY_LIBS="$RUBY_LIBS -L$rubylibdir"
!   elif test "$librubyarg" = "libruby.a"; then
!     dnl required on Mac OS 10.3 where libruby.a doesn't exist
!     librubyarg="-lruby"
!     RUBY_LIBS="$RUBY_LIBS -L$rubylibdir"
  	fi
  
  	if test "X$librubyarg" != "X"; then
*** ../vim-7.3.175/src/auto/configure	2011-05-05 17:23:58.000000000 +0200
--- src/auto/configure	2011-05-05 18:06:47.000000000 +0200
***************
*** 6218,6233 ****
    { $as_echo "$as_me:${as_lineno-$LINENO}: checking --with-ruby-command argument" >&5
  $as_echo_n "checking --with-ruby-command argument... " >&6; }
  
  # Check whether --with-ruby-command was given.
  if test "${with_ruby_command+set}" = set; then :
!   withval=$with_ruby_command; RUBY_CMD="$withval"; { $as_echo "$as_me:${as_lineno-$LINENO}: result: $RUBY_CMD" >&5
  $as_echo "$RUBY_CMD" >&6; }
  else
    RUBY_CMD="ruby"; { $as_echo "$as_me:${as_lineno-$LINENO}: result: defaulting to $RUBY_CMD" >&5
  $as_echo "defaulting to $RUBY_CMD" >&6; }
  fi
  
- 
    # Extract the first word of "$RUBY_CMD", so it can be a program name with args.
  set dummy $RUBY_CMD; ac_word=$2
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
--- 6218,6233 ----
    { $as_echo "$as_me:${as_lineno-$LINENO}: checking --with-ruby-command argument" >&5
  $as_echo_n "checking --with-ruby-command argument... " >&6; }
  
+ 
  # Check whether --with-ruby-command was given.
  if test "${with_ruby_command+set}" = set; then :
!   withval=$with_ruby_command; RUBY_CMD="$withval"; vi_cv_path_ruby="$withval"; { $as_echo "$as_me:${as_lineno-$LINENO}: result: $RUBY_CMD" >&5
  $as_echo "$RUBY_CMD" >&6; }
  else
    RUBY_CMD="ruby"; { $as_echo "$as_me:${as_lineno-$LINENO}: result: defaulting to $RUBY_CMD" >&5
  $as_echo "defaulting to $RUBY_CMD" >&6; }
  fi
  
    # Extract the first word of "$RUBY_CMD", so it can be a program name with args.
  set dummy $RUBY_CMD; ac_word=$2
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
***************
*** 6292,6308 ****
  	  RUBY_LIBS="$rubylibs"
  	fi
  	librubyarg=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG["LIBRUBYARG"])'`
! 	if test -f "$rubyhdrdir/$librubyarg"; then
! 	  librubyarg="$rubyhdrdir/$librubyarg"
! 	else
! 	  rubylibdir=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG["libdir"])'`
! 	  if test -f "$rubylibdir/$librubyarg"; then
! 	    librubyarg="$rubylibdir/$librubyarg"
! 	  elif test "$librubyarg" = "libruby.a"; then
! 	    	    librubyarg="-lruby"
! 	  else
! 	    librubyarg=`$vi_cv_path_ruby -r rbconfig -e "print '$librubyarg'.gsub(/-L\./, %'-L#{Config.expand(Config::CONFIG[\"libdir\"])}')"`
! 	  fi
  	fi
  
  	if test "X$librubyarg" != "X"; then
--- 6292,6305 ----
  	  RUBY_LIBS="$rubylibs"
  	fi
  	librubyarg=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG["LIBRUBYARG"])'`
! 	librubya=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG["LIBRUBY_A"])'`
!   rubylibdir=`$vi_cv_path_ruby -r rbconfig -e 'print Config.expand(Config::CONFIG["libdir"])'`
! 	if test -f "$rubylibdir/$librubya"; then
! 	  librubyarg="$librubyarg"
!     RUBY_LIBS="$RUBY_LIBS -L$rubylibdir"
!   elif test "$librubyarg" = "libruby.a"; then
!         librubyarg="-lruby"
!     RUBY_LIBS="$RUBY_LIBS -L$rubylibdir"
  	fi
  
  	if test "X$librubyarg" != "X"; then
*** ../vim-7.3.175/src/version.c	2011-05-05 17:32:40.000000000 +0200
--- src/version.c	2011-05-05 18:08:52.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     176,
  /**/

-- 
"I simultaneously try to keep my head in the clouds and my feet on the
ground.  Sometimes it's a stretch, though."              -- Larry Wall

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
